packages/react-reconciler/src/ReactProfilerTimer.js

```diff
@@ -42,155 +35,47 @@ function recordCommitTime(): void {
-    now() - ((fiber.actualDuration: any): number) - totalElapsedPauseTime;
-  fiber.actualStartTime = now();
-}
-
-function pauseActualRenderTimerIfRunning(): void {
-  if (!enableProfilerTimer) {
-    return;
-  }
-  if (timerPausedAt === 0) {
-    timerPausedAt = now();
-  }
-  if (syncCommitStartTime > 0) {
-    // Don't count the time spent processing sycn work,
-    // Against yielded async work that will be resumed later.
-    totalElapsedPauseTime += now() - syncCommitStartTime;
-    syncCommitStartTime = 0;
-  } else {
-    totalElapsedPauseTime = 0;
-  }
-}
-
-function recordElapsedActualRenderTime(fiber: Fiber): void {
+function startProfilerTimer(fiber: Fiber): void {
   if (!enableProfilerTimer) {
     return;
   }
-  if (__DEV__) {
-    warningWithoutStack(
-      fiber === fiberStack.pop(),
-      'Unexpected Fiber (%s) popped.',
-      getComponentName(fiber.type),
-    );
-  }
 
-  fiber.actualDuration =
-    now() - totalElapsedPauseTime - ((fiber.actualDuration: any): number);
-}
+  profilerStartTime = now();
 
-function resumeActualRenderTimerIfPaused(isProcessingSyncWork: boolean): void {
-  if (!enableProfilerTimer) {
-    return;
-  }
-  if (isProcessingSyncWork && syncCommitStartTime === 0) {
-    syncCommitStartTime = now();
-  }
-  if (timerPausedAt > 0) {
-    totalElapsedPauseTime += now() - timerPausedAt;
-    timerPausedAt = 0;
+  if (((fiber.actualStartTime: any): number) < 0) {
+    fiber.actualStartTime = now();
   }
 }
 
-function resetActualRenderTimerStackAfterFatalErrorInDev(): void {
+function stopProfilerTimerIfRunning(fiber: Fiber): void {
   if (!enableProfilerTimer) {
     return;
   }
-  if (__DEV__) {
-    fiberStack.length = 0;
-  }
+  profilerStartTime = -1;
 }
 
-/**
- * The "base" render time is the duration of the “begin” phase of work for a particular fiber.
- * This time is measured and stored on each fiber.
- * The time for all sibling fibers are accumulated and stored on their parent during the "complete" phase.
- * If a fiber bails out (sCU false) then its "base" timer is cancelled and the fiber is not updated.
- */
-
-let baseStartTime: number = -1;
-
-function recordElapsedBaseRenderTimeIfRunning(fiber: Fiber): void {
+function stopProfilerTimerIfRunningAndRecordDelta(
+  fiber: Fiber,
+  overrideBaseTime: boolean,
+): void {
   if (!enableProfilerTimer) {
     return;
   }
-  if (baseStartTime !== -1) {
-    fiber.selfBaseDuration = now() - baseStartTime;
-  }
-}
 
-function startBaseRenderTimer(): void {
-  if (!enableProfilerTimer) {
-    return;
-  }
-  if (__DEV__) {
-    if (baseStartTime !== -1) {
-      warningWithoutStack(
-        false,
-        'Cannot start base timer that is already running. ' +
-          'This error is likely caused by a bug in React. ' +
-          'Please file an issue.',
-      );
+  if (profilerStartTime >= 0) {
+    const elapsedTime = now() - profilerStartTime;
+    fiber.actualDuration += elapsedTime;
+    if (overrideBaseTime) {
```

Comment:
Base time is a little awkward to handle. Before, we were only including "begin" phase work in the base time value– and this PR continues that behavior, because I wasn't sure how to preserve the behavior we wanted otherwise (without adding an additional field).

The behavior is, essentially:
* When cloning a fiber, copy the base time from the previous render.
* If the fiber bails out, leave that value in place.
* If the fiber renders, override the base time value.
* When the fiber completes, bubble up base times from its children.

Since we are using a single timer to track both the "actual duration" and "base duration" values, and this timer wraps both the "begin" and "render" phases– I opted to only update base time during render (since we know definitively whether we have bailed out or not during that phase).

If this is silly for some reason, or you know of a better way to achieve this, let's chat!